addons/website_slides/controllers/thread.py

```diff
@@ -0,0 +1,24 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.controllers import thread
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_update_content(
+        self, message_id, body, attachment_ids, attachment_tokens=None, partner_ids=None, **kwargs
+    ):
+        if rating_value := kwargs.pop("rating_value", False):
+            domain = [("message_id", "=", message_id)]
+            rating = (
+                request.env["rating.rating"].sudo().search(domain, order="write_date DESC", limit=1)
+            )
+            rating.rating = rating_value
+            rating.feedback = html2plaintext(body)
+        return super().mail_message_update_content(
+            message_id, body, attachment_ids, attachment_tokens, partner_ids, **kwargs
+        )
```

Comment:
Shouldn't the rating override be in the rating module rather than on website_slides?

Reply:
This part of the widget (rating composer) is only used in this module. I thought about moving it to `portal_rating` because all the other related code to send the `rating_value` is there, but since the override was already in website_slides I kept it there. If we want to move it, isn't `portal_rating` a better choice?